src/librustc_typeck/diagnostics.rs

```diff
@@ -733,6 +748,38 @@ fn some_func(x: &mut i32) {
 ```
 "##,
 
+E0071: r##"
+You tried to use a structure initialization with a non-structure type.
+Example of erroneous code:
+
+```
+enum Foo { f };
+
+let u = Foo::f { value: 0i32 }; // error: Foo:f isn't a structure!
+// or even simpler:
+let u = t { random_field: 0i32 }; //error: t isn't a structure!
+```
+
+To fix this error, please declare the structure with the given name
```

Comment:
This isn't the fix. There's no way to "fix" this, really, one should use variant constructors for enums and struct constructors for structs; if the wrong one is being used, find out why and fix the struct/enum or the constructor.